### PR TITLE
GEOMESA-2085 Support for Kafka 1.0.0

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by jnh5y** [comment]
Does this change any dependencies?  Also, does anything need to change in GeoMesa-NiFi?

**Comment by elahrvivaz** [comment]
doesn't change any dependencies - you can just plug in kafka 1.0 at runtime now. Nifi shouldn't need any changes either.

**Comment by elahrvivaz** [comment]
oops, it did change one test dependency - pushed the new cq file

File: geomesa-kafka/geomesa-kafka-datastore/src/test/scala/org/locationtech/geomesa/kafka/EmbeddedKafka.scala [hunk 1]
```diff
@@ -33,6 +33,11 @@ class EmbeddedKafka extends Closeable {
 
   val brokers = s"127.0.0.1:${server.socketServer.boundPort()}"
 
+  // for kafka 1.0.0:
+  // import org.apache.kafka.common.network.ListenerName
+  // import org.apache.kafka.common.security.auth.SecurityProtocol
+  // val brokers = s"127.0.0.1:${server.socketServer.boundPort(ListenerName.forSecurityProtocol(SecurityProtocol.PLAINTEXT))}"
+
```
Comment (jnh5y) [inline geomesa-kafka/geomesa-kafka-datastore/src/test/scala/org/locationtech/geomesa/kafka/EmbeddedKafka.scala]: Hmm?  The test has to change with 1.0.0?!

File: pom.xml [hunk 2]
```diff
@@ -91,9 +91,10 @@
         <hbase.guava.version>12.0.1</hbase.guava.version>
         <cassandra.version>3.0.0</cassandra.version>
         <cassandra.guava.version>18.0</cassandra.guava.version>
-        <kafka.08.version>0.8.2.1</kafka.08.version>
-        <kafka.09.version>0.9.0.1</kafka.09.version>
-        <kafka.10.version>0.10.0.1</kafka.10.version>
+        <kafka.version>0.9.0.1</kafka.version> <!-- also supports: 0.10.2.1, 1.0.0 -->
+        <zkclient.version>0.7</zkclient.version> <!-- corresponding versions: 0.10, 0.10 -->
+        <kafka.jopt.version>3.2</kafka.jopt.version> <!-- corresponding versions: 4.9, 5.0.4 -->
```
Comment (jnh5y) [inline pom.xml]: Should we document how developers can manage all the versions?  I *think* I'd know what to do, but I doubt that it is clear to an outside developer.

File: geomesa-kafka/geomesa-kafka-datastore/src/test/scala/org/locationtech/geomesa/kafka/EmbeddedKafka.scala (hunk 1 above)
Comment (elahrvivaz) [reply geomesa-kafka/geomesa-kafka-datastore/src/test/scala/org/locationtech/geomesa/kafka/EmbeddedKafka.scala]: yeah, but in general you wouldn't ever need to compile with other versions of kafka - i just left it in there for the future

Comment (jnh5y) [reply geomesa-kafka/geomesa-kafka-datastore/src/test/scala/org/locationtech/geomesa/kafka/EmbeddedKafka.scala]: Roger that.

File: pom.xml (hunk 2 above)
Comment (elahrvivaz) [reply pom.xml]: you shouldn't actually need to change the versions, as it's a provided dep you plug in at runtime. but looks like the kakfa datastore docs need some updating, so I'll add it in there as part of the big docs pr

**Pull request merged**